pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -0,0 +1,317 @@
+
+	esv1alpha1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1alpha1"
+	smmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+	"github.com/external-secrets/external-secrets/pkg/provider"
+	"github.com/external-secrets/external-secrets/pkg/provider/schema"
+)
+
+// Provider satisfies the provider interface.
+type Provider struct{}
+
+// interface to keyvault.BaseClient.
+type SecretClient interface {
+	GetKey(ctx context.Context, vaultBaseURL string, keyName string, keyVersion string) (result keyvault.KeyBundle, err error)
+	GetSecret(ctx context.Context, vaultBaseURL string, secretName string, secretVersion string) (result keyvault.SecretBundle, err error)
+	GetSecretsComplete(ctx context.Context, vaultBaseURL string, maxresults *int32) (result keyvault.SecretListResultIterator, err error)
+}
+
+// Azure satisfies the provider.SecretsClient interface.
+type Azure struct {
+	kube       client.Client
+	store      esv1alpha1.GenericStore
+	baseClient SecretClient
+	vaultURL   string
+	namespace  string
+}
+
+func init() {
+	schema.Register(&Provider{}, &esv1alpha1.SecretStoreProvider{
+		AzureKV: &esv1alpha1.AzureKVProvider{},
+	})
+}
+
+// NewClient constructs a new secrets client based on the provided store.
+func (p *Provider) NewClient(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (provider.SecretsClient, error) {
+	return newClient(ctx, store, kube, namespace)
+}
+
+func newClient(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (provider.SecretsClient, error) {
+	anAzure := &Azure{
+		kube:      kube,
+		store:     store,
+		namespace: namespace,
+	}
+	azClient, vaultURL, err := anAzure.newAzureClient(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	anAzure.baseClient = azClient
+	anAzure.vaultURL = vaultURL
+	return anAzure, nil
+}
+
+// Implements store.Client.GetSecret Interface.
+// Retrieves a secret/Key/Certificate with the secret name defined in ref.Name
+// The Object Type is defined as a prefix in the ref.Name , if no prefix is defined , we assume a secret is required.
+func (a *Azure) GetSecret(ctx context.Context, ref esv1alpha1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	version := ""
+	objectType := "secret"
+	basicClient := a.baseClient
+	secretValue := ""
+
+	if ref.Version != "" {
+		version = ref.Version
+	}
+
+	secretName := ref.Key
+	nameSplitted := strings.Split(secretName, "/")
+
+	if len(nameSplitted) > 1 {
+		objectType = nameSplitted[0]
+		secretName = nameSplitted[1]
+		// Shall we neglect any later tokens or raise an error ??
+	}
+
+	switch objectType {
+	case "secret":
+		secretResp, err := basicClient.GetSecret(context.Background(), a.vaultURL, secretName, version)
+		if err != nil {
+			return nil, err
+		}
+		secretValue = *secretResp.Value
+
+	case "cert":
+		secretResp, err := basicClient.GetSecret(context.Background(), a.vaultURL, secretName, version)
```

Comment:
I think this should be [GetCertificate](https://pkg.go.dev/github.com/azure/azure-sdk-for-go@v31.1.0+incompatible/services/keyvault/v7.0/keyvault#BaseClient.GetCertificate) to retrieve an object with type certificate